pkg/cmd/pulumi/pulumi.go

```diff
@@ -713,7 +715,18 @@ type cachedVersionInfo struct {
 func getUpgradeMessage(latest semver.Version, current semver.Version, isDevVersion bool) string {
 	cmd := getUpgradeCommand(isDevVersion)
 
-	msg := fmt.Sprintf("A new version of Pulumi is available. To upgrade from version '%s' to '%s', ", current, latest)
+	// If the current version is "very old", we'll return a more urgent message. "Very old" is defined as any number of
+	// major versions behind, or more than 24 minor versions behind. Assuming a release cadence of on average 1 minor
+	// version per week, this translates to roughly 6 months.
+	prefix := "A new version of Pulumi is available."
+
+	sign, majorDiff, minorDiff := diffVersions(current, latest)
+	if sign < 0 && (majorDiff > 0 || minorDiff > 24) {
```

Comment:
We probably don't need/want to warn all users that strongly if we ever bump a major version.  Maybe only warn if `(majorDiff > 0 && minor > 24) || minorDiff > 24` or something like that?

Reply:
Hmm yeah that's tricky.  I do wonder if it would be better to not warn than have this strong warning that might be upsetting to people though. 

Although, I guess we should just be keeping the oldest allowed version to a 3.x version if we decide to upgrade to 4.x, which we can implement in the service.  And then what you have here is fine as it is.